resources/local.rb

```diff
@@ -0,0 +1,32 @@
+unified_mode true
+
+property :python_version, String, name_property: true
+property :path, String, required: true
+property :user, String
+
+# This sets the Local pyenv version
+# e.g. "pyenv local" should return the version we set
+
+action :create do
+  pyenv_script 'local' do
+    code "pyenv local #{new_resource.python_version}"
+    cwd new_resource.path
+    user new_resource.user if new_resource.user
```

Comment:
The `if` isn't required as Chef will just pass `nil` through to the inside resource anyway if user isn't set on the outside one